mlflow/johnsnowlabs/__init__.py

```diff
@@ -436,21 +440,26 @@ def _save_model_metadata(
 
     _PythonEnv.current().to_yaml(str(Path(dst_dir) / _PYTHON_ENV_FILE_NAME))
 
-    _save_jars_and_lic(dst_dir)
+    _save_jars_and_lic(dst_dir, gpu=gpu)
 
 
-def _save_jars_and_lic(dst_dir, store_license=False):
+def _save_jars_and_lic(dst_dir, store_license=False, gpu=False):
     from johnsnowlabs.auto_install.jsl_home import get_install_suite_from_jsl_home
     from johnsnowlabs.py_models.jsl_secrets import JslSecrets
+    from johnsnowlabs.utils.enums import JvmHardwareTarget
 
     deps_data_path = Path(dst_dir) / _JOHNSNOWLABS_MODEL_PATH_SUB / "jars.jsl"
     deps_data_path.mkdir(parents=True, exist_ok=True)
+    jar_type = JvmHardwareTarget(
+        "gpu" if gpu else "cpu"
+    )  # _JOHNSNOWLABS_ENV_ENABLE_GPU in os.environ
```

Comment:
There is no env variable `_JOHNSNOWLABS_ENV_ENABLE_GPU`.
To log a GPU model. users can call
`mlflow.johnsnowlabs.log_model(model, path, gpu=True)` or  `mlflow.johnsnowlabs.save(model, path, gpu=True)` .

The gpu=True value gets passed down the callstack. 

We require no env var for GPU, because the jar is public.
The Visual/Medical jars require a env var to resolve the jar location. 

I originally had the idea to use `_JOHNSNOWLABS_ENV_ENABLE_GPU` but dropped it because it requires no secret. This comment is a left over from that idea, removing it.